STACpopulator/extensions/cmip6.py

```diff
@@ -0,0 +1,356 @@
+import json
+from datetime import datetime
+from typing import (
+    Any,
+    Generic,
+    Iterable,
+    List,
+    Literal,
+    MutableMapping,
+    Type,
+    TypeVar,
+    Union,
+    Optional,
+    cast,
+    get_args
+)
+
+import pystac
+import pyessv
+from pydantic import AnyHttpUrl, BaseModel, ConfigDict, Field, FieldValidationInfo, field_validator
+from pydantic.fields import FieldInfo
+from pystac.extensions import item_assets
+from pystac.extensions.base import (
+    ExtensionManagementMixin,
+    PropertiesExtension,
+    SummariesExtension,
+)
+
+from STACpopulator.models import AnyGeometry
+from STACpopulator.stac_utils import ServiceType, collection2literal, ncattrs_to_bbox, ncattrs_to_geometry
+
+T = TypeVar("T", pystac.Collection, pystac.Item, pystac.Asset, item_assets.AssetDefinition)
+
+SchemaName = Literal["cmip6"]
+SCHEMA_URI: str = "https://stac-extensions.github.io/cmip6/v1.0.0/schema.json"
```

Comment:
This schema URI is not valid as there is no officially proposed STAC extension for CMIP6 yet. Tom's schema URI is "https://raw.githubusercontent.com/TomAugspurger/cmip6/main/json-schema/schema.json", but that too has issues (https://github.com/TomAugspurger/cmip6/pull/1) and until that is fixed, it's not a correct Schema and it will not validate the CMIP6 items. How do you want to proceed with this?

Reply:
Good point. I think it would be preferable to revert to the resolvable endpoint until the extension becomes officially managed by the stac-extensions endpoint.